controllers/toolchaincluster/toolchaincluster_controller.go

```diff
@@ -19,9 +21,10 @@ import (
 
 // Reconciler reconciles a ToolchainCluster object
 type Reconciler struct {
-	Client     client.Client
-	Scheme     *runtime.Scheme
-	RequeAfter time.Duration
+	Client                  client.Client
+	Scheme                  *runtime.Scheme
+	RequeAfter              time.Duration
+	overwriteRunHealthcheck func(ctx context.Context, lcl client.Client, rcl client.Client, rclset *kubeclientset.Clientset, tc *toolchainv1alpha1.ToolchainCluster, lgr logr.Logger) ([]toolchainv1alpha1.Condition, error)
```

Comment:
let's not introduce any "overwrite" logic; instead of it:
* convert the `runHealthcheck` method into a function 
* rename this parameter to `CheckHealth` or something like that
* either in main.go pass the `runHealthcheck` function as a value of the `CheckHealth` param of the Reconciler, or set it as the default value as part of the `SetupWithManager` method.

Reply:
Okay This would be easier, 
i used the overwrite for ease of testing though